provider/priority_resource_test.go

```diff
@@ -0,0 +1,276 @@
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "default", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccPriorityResource_update(t *testing.T) {
+	rSlug := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	rSlugUpdated := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		CheckDestroy:      testAccCheckPriorityResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPriorityResourceConfig_basic(rSlug),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckPriorityResourceExistsWithAttributes_basic("firehydrant_priority.test_priority"),
+					resource.TestCheckResourceAttrSet("firehydrant_priority.test_priority", "id"),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "slug", fmt.Sprintf("TESTPRIORITY%s", rSlug)),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "default", "false"),
+				),
+			},
+			{
+				Config: testAccPriorityResourceConfig_update(rSlugUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPriorityResourceExistsWithAttributes_update("firehydrant_priority.test_priority"),
+					resource.TestCheckResourceAttrSet("firehydrant_priority.test_priority", "id"),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "slug", fmt.Sprintf("TESTPRIORITY%s", rSlugUpdated)),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "description", fmt.Sprintf("test-description-%s", rSlugUpdated)),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "default", "true"),
+				),
+			},
+			{
+				Config: testAccPriorityResourceConfig_basic(rSlugUpdated),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckPriorityResourceExistsWithAttributes_basic("firehydrant_priority.test_priority"),
+					resource.TestCheckResourceAttrSet("firehydrant_priority.test_priority", "id"),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "slug", fmt.Sprintf("TESTPRIORITY%s", rSlugUpdated)),
+					resource.TestCheckResourceAttr(
+						"firehydrant_priority.test_priority", "default", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccPriorityResource_validateSchemaAttributesSlug(t *testing.T) {
+	rSlug := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccPriorityResourceConfig_slugTooLong(rSlug),
+				ExpectError: regexp.MustCompile(`expected length of slug to be in the range (0 - 23)`),
```

Comment:
Minor syntax error on my part here 😅 
The tests failed because the parens need to be escaped:
```suggestion
				ExpectError: regexp.MustCompile(`expected length of slug to be in the range \(0 - 23\)`),
```